Allow to exit Town Settings without applying changes

ihhub opened the pull request:
The previous implementation did not allow map makers to exit Town Settings dialog discarding changes. Having 2 normal buttons in the dialog is very problematic and this is what I achieved with them:
![image](https://github.com/user-attachments/assets/85b29ad1-8c7f-4f6f-aa0c-204b04bdf8f1)

As you can see the buttons are too close to each other and borders. Hence, we decided to use small buttons alongside the status bar:
![image](https://github.com/user-attachments/assets/eee96cd1-a32b-4e9e-91be-04ddc9a55539)

Districh-ru commented on src/fheroes2/editor/editor_castle_details_window.cpp:
```diff
@@ -588,11 +597,18 @@ namespace Editor
+                }
+            }
+            else if ( le.isMouseCursorPosInArea( buttonCancel.area() ) ) {
+                message = _( "Exit this menu without doing anything." );
```
Hi @ihhub and @zenseii, 
what do you think of the next rephrasing?
```suggestion
                message = _( "Exit this dialog, discarding the changes made." );
```

ihhub commented:
Hi @zenseii , sure, I can give it a try.

ihhub commented:
Hi @zenseii and @Districh-ru , I reworked this pull request. Now we have 2 buttons alongside the status bar (see the updated description of this pull request). However, some original sentences are too short to be displayed fully in the status bar:
![image](https://github.com/user-attachments/assets/f1d7d0d6-47f4-4d23-9125-b75632b5e9f4)

We need to rework the sentences we use and possibly the UI to avoid mentioning and using right-clicking actions. We could use very small buttons like in events.

Districh-ru commented:
Hi @ihhub, sadly but not all phrases (including translations) are now fully shown in the status bar and if they are long they overlap the decorations:
![изображение](https://github.com/user-attachments/assets/8783d40e-6cb9-4ac1-936c-2dd04b0425bc)

Districh-ru commented:
> @Districh-ru , one task has been completed: I've reduced the text rendering area to avoid text be over decorations. By the way, this issue is also present in other dialogs.

Great, @ihhub!

We can open an issue and fix it in a new PR.

ihhub commented:
Hi @Districh-ru and @zenseii , I've addressed all comments here. The only thing is that right clicking on army resets a slot. We could fix it separately.

Districh-ru commented on src/fheroes2/editor/editor_castle_details_window.cpp:
```diff
@@ -576,6 +623,17 @@ namespace Editor
+                if ( le.MouseClickLeft( buttonResetArmy.area() ) ) {
+                    castleArmy.Reset( false );
+                    armyBar.Redraw( display );
+                    display.render( dialogRoi );
```
For this action we can render only the cleared army bar:
```suggestion
                    display.render( armyBar.GetArea() );
```

Districh-ru reviewed (approved):
Well done, @ihhub!
I left one minor optimization suggestion.
